core/libraries/rest_api/ModelDataTranslator.php

```diff
@@ -382,223 +385,395 @@ public static function prepareConditionsQueryParamsForModels(
-                            // but do verify it at least doesn't have any serialized data
-                            ModelDataTranslator::throwExceptionIfContainsSerializedData($query_param_value[1]);
-                            $translated_value[] = $query_param_value[1];
-                        } elseif (array_key_exists($op, $model->valid_null_style_operators())
-                                  && ! isset($query_param_value[1])) {
-                            // no arguments should have been provided, so don't look for any
-                        } elseif (isset($query_param_value[1])
-                                  && ! isset($query_param_value[2])
-                                  && ! array_key_exists(
-                                      $op,
-                                      array_merge(
-                                          $model->valid_in_style_operators(),
-                                          $model->valid_null_style_operators(),
-                                          $model->valid_like_style_operators(),
-                                          $model->valid_between_style_operators()
-                                      )
-                                  )
-                        ) {
-                            // it's a valid operator, but none of the exceptions. Treat it normally.
-                            $translated_value[] = ModelDataTranslator::prepareFieldValuesFromJson(
-                                $field,
-                                $query_param_value[1],
-                                $requested_version,
-                                $timezone
-                            );
-                        } else {
-                            // so they provided a valid operator, but wrong number of arguments
-                            if (defined('EE_REST_API_DEBUG_MODE') && EE_REST_API_DEBUG_MODE) {
-                                throw new RestException(
-                                    'wrong_number_of_arguments',
-                                    sprintf(
-                                        esc_html__(
-                                            'The operator you provided, "%1$s" had the wrong number of arguments',
-                                            'event_espresso'
-                                        ),
-                                        $op
-                                    ),
-                                    array(
-                                        'status' => 400,
+                    // they're doing something like "&where[EVT_ID][IN]=1,2,3" or "&where[EVT_ID][>]=5"
+                    if (array_key_exists(
+                        $sub_array_key,
+                        array_merge(
+                            $model->valid_in_style_operators(),
+                            $model->valid_between_style_operators()
+                        )
```

Comment:
this will get called about once per request parameter on a request. So, usually it will get called about 2-5 times per request. If it were being called dozens of times per request, or if those arrays could be big, I think that'd be great. These arrays are just 2-5 items each, so merging the arrays 2-5 times on a request seems reasonable, and avoids a little extra complexity. Thoughts?

Reply:
oh ya that's fine... i didn't think the arrays were going to be that big (how many operators are can there be?) but wasn't sure how many times they were getting merged per request. This is fine then.